app/assets/javascripts/discourse/app/initializers/topic-footer-buttons.js

```diff
@@ -81,27 +83,26 @@ export default {
       },
       label() {
         if (!this.topic.isPrivateMessage || this.site.mobileView) {
-          const bookmarkedPosts = this.topic.bookmarked_posts;
-          const bookmarkedPostsCount = bookmarkedPosts
-            ? bookmarkedPosts.length
-            : 0;
-
-          if (bookmarkedPostsCount === 0) {
+          if (this.topic.bookmarkCount === 0) {
             return "bookmarked.title";
-          } else if (bookmarkedPostsCount === 1) {
+          } else if (this.topic.bookmarkCount === 1) {
             return "bookmarked.edit_bookmark";
           } else {
             return "bookmarked.clear_bookmarks";
           }
         }
       },
       translatedTitle() {
-        const bookmarkedPosts = this.topic.bookmarked_posts;
-        if (!bookmarkedPosts || bookmarkedPosts.length === 0) {
+        const bookmarks = this.topic.bookmarks;
+        if (!bookmarks || this.topic.bookmarkCount === 0) {
```

Comment:
If we implement checking if bookmarks is null in the computed property, we may have here just
```javascript
if (this.topic.bookmarkCount === 0) {
...
```

Reply:
Good point, will tweak this then merge :+1: